src/bookshelf/src/bksw_sql.erl

```diff
@@ -0,0 +1,253 @@
+    end.
+
+%% @doc bookshelf queries.
+%% Referenced by sys.config in {sqerl, [{ prepared_staements...
+statements(_) ->
+    Path = filename:join([code:priv_dir(bookshelf), "pgsql_statements.config"]),
+    {ok, Statements} = file:consult(Path),
+    Statements.
+
+create_bucket(BucketName) ->
+    case sqerl:statement(insert_bucket, [BucketName], count) of
+        {ok, 1} ->
+            ok;
+        {conflict, _} ->
+            error("Never should happen");
+        Error ->
+            Error
+    end.
+
+-spec create_bucket_b(binary()) -> boolean().
+create_bucket_b(Bucket) ->
+    case bksw_sql:find_bucket(Bucket) of
+        {ok, none} ->
+            case bksw_sql:create_bucket(Bucket) of
+                {ok, 1} ->
+                    true;
+                _Error ->
+                    false
+            end;
+        {ok, _} ->
+            true
+    end.
+
+delete_bucket(BucketName) ->
+    case sqerl:statement(delete_bucket, [BucketName], count) of
+        {ok, 1} ->
+            ok;
+        Error ->
+            Error
+    end.
+
+-spec delete_bucket_b(binary()) -> boolean().
+delete_bucket_b(Bucket) ->
+    case bksw_sql:delete_bucket(Bucket) of
+        ok ->
+            true;
+        _ ->
+            false
+    end.
+
+find_bucket(BucketName) ->
+    case sqerl:select(find_bucket, [BucketName], first_as_scalar, [bucket_id]) of
+        {ok, BucketId} ->
+            {ok, BucketId};
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+-spec bucket_exists(binary()) -> boolean().
+bucket_exists(Bucket) ->
+    case bksw_sql:find_bucket(Bucket) of
+        {ok, none} ->
+            false;
+        {ok, _} ->
+            true
+end.
+
+list_buckets() ->
+    case sqerl:select(list_buckets, [], rows_as_records, ?DB_BUCKET_TX_FM) of
+        {ok, none} ->
+            [];
+        {ok, Buckets} ->
+            Buckets;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+list_bucket(BucketName) ->
+    case sqerl:select(list_bucket_with_hash, [BucketName], rows_as_records, ?DB_FILE_TX_FM) of
+        {ok, Buckets} ->
+            Buckets;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+create_file_link_data(BucketName, Name, DataId) ->
+    case sqerl:select(create_file_link_data, [BucketName, Name, DataId], first_as_scalar, [success]) of
+        {ok, FileId} ->
+            {ok, FileId};
+        {conflict, Reason} ->
+            {error, Reason};
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+link_file_data(FileId, DataId) ->
+    case sqerl:select(link_file_data, [FileId, DataId], first_as_scalar, [success]) of
+        {ok, _} ->
+            ok;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+find_file(Bucket, Name) ->
+    case sqerl:select(find_file, [Bucket, Name], {first_as_record, ?DB_FILE_TX_FM }) of
+        %% Awkward sanity check that we got back the expected record type here.
```

Comment:
I think we can nuke these. THere's no value in them, the record must match what we give it based on the way sqerl assembles the response.